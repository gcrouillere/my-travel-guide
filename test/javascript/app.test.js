import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { MemoryRouter } from 'react-router'
import { BrowserRouter as Router, Route } from 'react-router-dom'

import App from '../../app/javascript/packs/app';
import Article from '../../app/javascript/packs/article';
import ArticleForm from '../../app/javascript/packs/articleForm';

configure({ adapter: new Adapter() });

 describe('App test suite', () => {
  it('renders App without crashing', () => {
    const AppTree = renderer.create(<Router><App/></Router>).toJSON()
    expect(AppTree).toMatchSnapshot()
  });

  it('renders Article according to path', () => {
    const AppTreeArticle = mount(
      <MemoryRouter initialEntries={["/articles/4"]} initialIndex={0} >
        <App/>
      </MemoryRouter>)
    expect(AppTreeArticle.find(Article).length).toEqual(1)
  });

  it('renders ArticleForm according to path', () => {
    const AppTreeArticleNew = mount(
      <MemoryRouter initialEntries={["/articles/new"]} initialIndex={0}>
        <App/>
      </MemoryRouter>)
    expect(AppTreeArticleNew.find(ArticleForm).length).toEqual(1)
  });
 })
