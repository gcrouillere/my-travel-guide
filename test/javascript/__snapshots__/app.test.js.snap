// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders App without crashing 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-light bg-dark"
    >
      <a
        className="navbar-brand nav-link text-white"
        href="/"
        onClick={[Function]}
      >
        My Travel Guide
      </a>
      <a
        className="nav-link text-white"
        href="/articles/new"
        onClick={[Function]}
      >
        Create article
      </a>
    </nav>
  </div>
  <div
    className="container articles-container"
  >
    <div
      className="row"
    />
  </div>
</div>
`;
